mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala

```diff
@@ -122,7 +127,74 @@ trait MesosClient {
 object MesosClient extends StrictLogging with StrictLoggingFlow {
   case class MesosRedirectException(leader: URL) extends Exception(s"New mesos leader available at $leader")
 
-  case class ConnectionInfo(url: URL, streamId: String)
+  case class Session(url: URL, streamId: String, credentials: Option[CredentialsProvider] = None) {
+    lazy val isSecured: Boolean = url.getProtocol == "https"
+    lazy val port = if (url.getPort == -1) url.getDefaultPort else url.getPort
+
+    def createPostRequest(bytes: Array[Byte], maybeCredentials: Option[HttpCredentials]): HttpRequest =
+      HttpRequest(
+        HttpMethods.POST,
+        uri = Uri(s"${url.getPath}/api/v1/scheduler"),
+        entity = HttpEntity(MesosClient.ProtobufMediaType, bytes),
+        headers = MesosClient.MesosStreamIdHeader(streamId) :: maybeCredentials.map(Authorization(_)).toList
+      )
+
+    case class SessionFlow() extends GraphStage[FlowShape[Array[Byte], HttpRequest]] {
+
+      private val callsInlet = Inlet[Array[Byte]]("specs")
+      private val requestsOutlet = Outlet[HttpRequest]("effects")
+      override val shape = FlowShape.of(callsInlet, requestsOutlet)
+
+      override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = {
+        new GraphStageLogic(shape) {
```

Comment:
There is something wrong here since we only process one call. @timcharper and @meln1k, any ideas?